src/main/java/dal/fatih/todoly/Main.java

```diff
@@ -55,30 +54,57 @@ public static void listAllTasks() {
             }
         }
     }
+
     public static void showTaskDetails() {
         System.out.print("Task Id :");
         String taskId = scn.nextLine();
-        Task task=tasks.get(taskId);
+        Task task = tasks.get(taskId);
         if (task != null) {
             System.out.println(tasks.get(taskId));
         } else {
             System.out.println("Task not found");
         }
     }
-    public static void deleteTask(){
+
+    public static void deleteTask() {
         System.out.print("Task Id :");
         String taskId = scn.nextLine();
-        Task task=tasks.get(taskId);
+        Task task = tasks.get(taskId);
         if (task != null) {
             tasks.remove(taskId);
-            System.out.println(task.getTitle() +"titled task deleted");
+            System.out.println(task.getTitle() + "titled task deleted");
         } else {
             System.out.println("Task not found");
         }
     }
-    
-    static Scanner scn = new Scanner(System.in);
-    static Map<String, Task> tasks = new HashMap<String, Task>();
+
+    public static void filterTask() {
+
+        try {
+            Map<String, Task> taskDate = tasks;
+            List<Task> foundTask = new ArrayList<>();
+            System.out.println("Last Date");
+            String lastDateInput = scn.nextLine();
+            Date lastDate = dueDateParser.parse(lastDateInput);
+
+            for (Task task : taskDate.values()) {
+                if (task.getDate().before(lastDate)) {
+                    foundTask.add(task);
+                }
+            }
+            if (foundTask.isEmpty()) {
+                System.out.println("No tasks found between these dates");
+            } else {
+                System.out.println(foundTask);
+            }
+        } catch (Exception e) {
+            System.out.println("Incorrect date format");
+        }
+    }
+
+    private static Scanner scn = new Scanner(System.in);
+    private static Map<String, Task> tasks = new HashMap<String, Task>();
+    private static final DateFormat dueDateParser = new SimpleDateFormat("dd/MM/yyyy");

```

Comment:
Move these fields to be at the beginning of the class. The fIelds shouldn't be here somewhere in the middle. Apart from that, define all of them to be final to be consistent.